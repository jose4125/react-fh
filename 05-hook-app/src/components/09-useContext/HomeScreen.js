import { useContext } from "react";
import { UserContext } from "./UserContext";

export const HomeScreen = () => {
  const { user } = useContext(UserContext);
  return (
    <div>
      <h1>home</h1>
      <pre>{JSON.stringify(user, null, 2)}</pre>
    </div>
  );
};
