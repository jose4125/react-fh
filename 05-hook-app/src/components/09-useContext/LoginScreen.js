import { useContext } from "react";
import { UserContext } from "./UserContext";

export const LoginScreen = () => {
  const { setUser } = useContext(UserContext);
  return (
    <div>
      <h1>login</h1>
      <hr />
      <button onClick={() => setUser({ id: 2, name: "jose" })}>Login</button>
    </div>
  );
};
