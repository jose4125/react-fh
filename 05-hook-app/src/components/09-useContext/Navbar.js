import { NavLink } from "react-router-dom";

export const NavBar = () => {
  return (
    <nav>
      <ul>
        <li>
          <NavLink exact activeClassName="active" to="/">
            Home
          </NavLink>
        </li>
        <li>
          <NavLink activeClassName="active" to="/about">
            About
          </NavLink>
        </li>
        <li>
          <NavLink activeClassName="active" to="/login">
            Login
          </NavLink>
        </li>
      </ul>
    </nav>
  );
};
