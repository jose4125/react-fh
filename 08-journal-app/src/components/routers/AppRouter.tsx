import { useEffect, useState } from 'react';
import { firebase } from '../firebase/firebase-config';
import { BrowserRouter as Router, Switch, Redirect } from 'react-router-dom';
import AuthRouter from './AuthRouter';
import { useDispatch } from 'react-redux';
import { login } from '../auth/authActions';
import JournalScreen from '../journalScreen/JournalScreen';
import PrivateRoute from './PrivateRoute';
import PublicRoute from './PublicRoute';

const AppRouter = (): JSX.Element => {
  const dispatch = useDispatch();
  const [isGettingUserInfo, setIsGettingUserInfo] = useState(true);
  const [isAuthenticated, setIsAuthenticated] = useState(false);

  useEffect(() => {
    firebase.auth().onAuthStateChanged(user => {
      if (user?.uid) {
        dispatch(login({ uid: user.uid, displayName: user.displayName }));
        setIsAuthenticated(true);
      }

      if (!user?.uid) {
        setIsAuthenticated(false);
      }
      setIsGettingUserInfo(false);
    });
  }, [dispatch, setIsGettingUserInfo, setIsAuthenticated]);

  if (isGettingUserInfo) {
    return <h1>Getting info...</h1>;
  }

  return (
    <Router>
      <div>
        <Switch>
          <PrivateRoute isAuthenticated={isAuthenticated} exact path="/">
            <JournalScreen />
          </PrivateRoute>
          <PublicRoute isAuthenticated={isAuthenticated} path="/auth">
            <AuthRouter />
          </PublicRoute>
          <Redirect to="/auth/login" />
        </Switch>
      </div>
    </Router>
  );
};

export default AppRouter;
