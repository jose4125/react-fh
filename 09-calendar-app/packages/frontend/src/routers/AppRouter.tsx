import { ReactElement } from 'react';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import { Calendar } from '../components/Calendar/Calendar';
import { LoginRegister } from '../components/LoginRegister/LoginRegister';

export const AppRouter = (): ReactElement => {
  return (
    <Router>
      <Routes>
        <Route path="/" element={<Calendar />} />
        <Route path="login" element={<LoginRegister />} />
        <Route path="*" element={<Calendar />} />
      </Routes>
    </Router>
  );
};
