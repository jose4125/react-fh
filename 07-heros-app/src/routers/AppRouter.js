import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import LoginScreen from "../components/login/LoginScreen";
import DashboardRouter from "./DashboardRouter";

const AppRouter = () => {
  return (
    <Router>
      <div>
        <Switch>
          <Route exact path="/login">
            <LoginScreen />
          </Route>
          <Route path="/">
            <DashboardRouter />
          </Route>
        </Switch>
      </div>
    </Router>
  );
};

export default AppRouter;
