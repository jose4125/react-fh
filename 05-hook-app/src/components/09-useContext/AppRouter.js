import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect,
} from "react-router-dom";
import { AboutScreen } from "./AboutScreen";
import { HomeScreen } from "./HomeScreen";
import { LoginScreen } from "./LoginScreen";
import { NavBar } from "./Navbar";

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />
        <Switch>
          <Route exact path="/" component={HomeScreen} />
          <Route path="/about" component={AboutScreen} />
          <Route path="/login" component={LoginScreen} />
          <Redirect to="/" />
        </Switch>
      </div>
    </Router>
  );
};
