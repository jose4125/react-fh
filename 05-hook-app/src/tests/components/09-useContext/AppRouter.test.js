import { mount } from "enzyme";
import { AppRouter } from "../../../components/09-useContext/AppRouter";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe("test AppRouter", () => {
  const wrapper = mount(
    <UserContext.Provider
      value={{
        user: {},
      }}
    >
      <AppRouter />
    </UserContext.Provider>
  );

  test("Should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
