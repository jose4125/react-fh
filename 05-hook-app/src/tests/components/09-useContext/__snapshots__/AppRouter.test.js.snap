// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test AppRouter Should render properly 1`] = `
<div>
  <nav>
    <ul>
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
    </ul>
  </nav>
  <div>
    <h1>
      home
    </h1>
    <pre>
      {}
    </pre>
  </div>
</div>
`;
